sssd_test_framework/utils/authentication.py

```diff
@@ -413,23 +412,29 @@ def __init__(self, host: MultihostHost) -> None:
         self.opts = "-o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no"
         """SSH CLI options."""
 
-    def password(self, username: str, password: str) -> bool:
+    def password(self, username: str, password: str, output: bool = False) -> bool | str:
         """
         SSH to the remote host and authenticate the user with password.
 
-        :param name: User name.
-        :type name: str
+        :param username: Username.
+        :type username: str
         :param password: User password.
         :type password: str
-        :return: True if authentication was successful, False otherwise.
-        :rtype: bool
+        :param output: Log output.
+        :type output: bool
+        :return: True if authentication was successful, False otherwise. If output is true, return output.
```

Comment:
Wouldn't it be better to return the two values (auth success/failure and output)? Is that possible in python?

Reply:
It's not used very often and what tests will use it isn't concerned about it passing.  Wanted to ensure that no code needed to change from this PR.

Reply:
Create method `password_with_output`, that returns `tuple[bool, str, str]` (result, stdout, stderr) so you get all information. `password` will call `password_with_output`.

This way, it is clear what is return, you get all information that you might need and it will remain type safe.